### PR TITLE
Add proper wasp nests

### PR DESCRIPTION
**Pull request by Venera3**
#### Summary
Content "Add large wasp nests"

#### Purpose of change

After #47960 wasps should hopefully be in a better spot balance-wise, but they're sorely missing a proper non-map extra location. Thankfully, there are a lot of high, lofty radio towers around for them to make a home in.

#### Describe the solution

Add a new variant to both out of town radio tower locations with a large wasp nest built overlapping with the structure. It spawns foragers in a 3-10 overmap tile range and houses a queen to replenish the local numbers. Also added a random chance of normal radio towers to spawn a single queen that's sizing the site up to build a nest.

- [x] Adjust spawn chance / variant radio tower chance: 80% per overmap each for the large nests, with a slightly higher city minimum distance to keep them from overlapping a city too much. The normal radio tower with a queen spawns half as often as the sniper chunk (about 5% of the time).
 
#### Describe alternatives you've considered

Have them as non-unique specials, but about half of the current radio tower spawns being replaced by a faceful of wasps sounded a bit too metal, even for me. The 80% per overmap chance is consistent with anthills/beehives, and unlike those a lot of players seek out radio towers for a free sniper rifle.
Add them  as a variant to the existing radio towers, but to do the nest justice I needed more than one Z-level. Add an embryonic nest to the normal radio tower chunk, but it wouldn't look half as nice as the real deal.

#### Testing

Specials spawn, have more or less the desired populations around them. Some more fine-tuning is still needed.

#### Additional context

Wasp/hornet nests are structured vertically, with "wafers" of cells hanging from the ceiling that's pretty unintuitive to mirror in game, but that's life for you. Beehives will look a tad more sane, for reference.
Depends on #47960 for the critters. Part of #47275 .
<details>
<summary>Screenshots</summary>

Approach

Nearing a nest, some foragers milling about and a weird shadow should clue you in to something being amiss

Level1

Surface level, if the bug guts strewn about and the battered building doesn't make you bug out you really deserve to become dinner.

Entrance

Small landing with a lot of prey remains (you have a chance of military/science loot, but nothing major).

Intermediate level

Lower cell layer 

Cell roof with the grumpy queen

Upper cell layer
</details>
Reference pics of real nests

https://i.pinimg.com/736x/05/93/18/0593189807de359641a897ca1a6eeec8.jpg
An overview diagram, the left part shows a ground-dwelling yellowjacket nest but the basic structure is the same among all eusocial species. 

https://www.flickr.com/photos/46573840@N07/4766945271/in/album-72157624309430975/
A schematic diagram, part of a beautiful album of wasp-inspired architecture.

### CODE REVIEW

**Comment by actual-nh** [comment]
Nice! Where/who are the (very nice) drawings from, BTW?

**Comment by Venera3** [comment]
> 
> 
> Nice! Where/who are the (very nice) drawings from, BTW?

Heh, that was a deserved rabbit hole (everything unappropriated, as usual). The first is from a paleoecology paper from 2002 , the pic itself is a combination/modification of an earlier one in Spadberry's Wasps (1973) for the left side and Wenzel, J.W., 1991. Evolution of Nest Architecture (1991). I found it on what appears to be a french Pathfinder forum about (I think) where one would meet giant insects, great minds and all that. The second is an artist's thesis work from 2010, updated both with slightly cleaner links.

File: data/json/mapgen_palettes/wasp_palette.json [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    "type": "mapgen",
+    "method": "json",
+    "nested_mapgen_id": "wasp_brood",
+    "//": "Egg cell.",
+    "weight": 200,
+    "object": {
+      "mapgensize": [ 1, 1 ],
+      "rows": [ "0" ],
+      "terrain": { "0": "t_floor_brood_fake" },
+      "item": { "0": { "item": "egg_wasp" } }
```
Comment (casswedson) [inline data/json/mapgen_palettes/wasp_palette.json:42]: It seems like `egg_wasp` is not defined in this this pr or defined in the `data` folder, weirdly it is mentioned here <https://github.com/CleverRaven/Cataclysm-DDA/blob/58d6d54fe588daa974a57e922f7f7810f794a831/gfx/RetroDaysTileset/tile_config.json#L2553>

Comment (actual-nh) [reply data/json/mapgen_palettes/wasp_palette.json:42]: Odd re that tile_config; `egg_wasp` might be defined in a "prerequisite" PR, #47960.

Comment (casswedson) [reply data/json/mapgen_palettes/wasp_palette.json:42]: Sorry, didn't read all the important parts >.<

Comment (actual-nh) [reply data/json/mapgen_palettes/wasp_palette.json:42]: In my experience, most people frequently miss those...

**Comment by actual-nh** [comment]
@BrettDong or whoever else might be doing labels - this PR depends on #47960, which has both String Freeze and Content Freeze, so should (unfortunately) be marked as Content Freeze.

File: data/json/overmap/overmap_terrain/overmap_terrain.json [hunk 2]
```diff
@@ -284,7 +284,7 @@
     "color": "light_gray",
     "see_cost": 2,
     "mondensity": 2,
-    "spawns": { "group": "GROUP_WASP_GUARD", "population": [ 5, 10 ] }
+    "spawns": { "group": "GROUP_WASP_GUARD", "population": [ 3, 8 ], "chance":100 }
```
Comment (actual-nh) [inline data/json/overmap/overmap_terrain/overmap_terrain.json:287]: `"chance": 100` (note space after the colon, BTW) is needed when there's only one thing spawning?

Comment (Venera3) [reply data/json/overmap/overmap_terrain/overmap_terrain.json:287]: Yyyup. That goes on the monster spawning weirdness list.

Comment (actual-nh) [reply data/json/overmap/overmap_terrain/overmap_terrain.json]: Thank you, I'll have to remember that... sigh.

File: data/json/mapgen_palettes/wasp_palette.json (hunk 1 above)
Comment (BrettDong) [inline data/json/mapgen_palettes/wasp_palette.json:42]: Did you define the item `egg_wasp`?

Comment (Venera3) [reply data/json/mapgen_palettes/wasp_palette.json:42]: Only in the other wasp PR. I'm a bit paranoid about accidentally overwriting some stuff if I decide to fiddle with them on the other branch.

**Pull request merged**